### PR TITLE
CLOUDP-187063: Override agent for Dockerized AtlasCLI

### PR DESCRIPTION
**Pull request by nicohrubec**
## Proposed changes

Adds a host parameter to the UserAgent, which defaults to "native". If the CLI is run from the docker container the host parameter is set to "container".

Jira ticket: CLOUDP-187063
Closes: CLOUDP-187063

## Checklist

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

## Further comments

Testing was done manually by using a local throwaway pipeline.

### CODE REVIEW

File: internal/store/store.go
Comment (matt-condon) [inline internal/store/store.go:287]: From what I can see, the UserAgent is set from 
https://github.com/mongodb/mongodb-atlas-cli/blob/master/internal/oauth/oauth.go#L70C11-L70C11

I wonder would it be better to pass the value from there, since it should only have to be called once when authenticating the CLI

File: internal/store/store.go
Comment (matt-condon) [inline internal/store/store.go:275]: Since there's a space rather than a dash in the existing pattern, I think we should consider doing the same here to be consistent (left a comment on the ticket also)
https://github.com/mongodb/mongodb-atlas-cli/blob/master/internal/config/profile.go#L76

image

Comment (nicohrubec) [reply internal/store/store.go]: Addressed

File: internal/store/store.go
Comment (nicohrubec) [reply internal/store/store.go]: I agree it seems like the better place to put it, updated

File: internal/oauth/oauth.go
Comment (matt-condon) [inline internal/oauth/oauth.go:68]: Per @drinkbird comment on the JIRA ticket 

> Thanks for flagging this. By doing some further reading, I see that what comes after "product/version" is "(comment)". We can extend what we have to include the environment in all cases.

Source: https://www.rfc-editor.org/rfc/rfc9110#name-parameters

Comment (nicohrubec) [reply internal/oauth/oauth.go]: updated

File: internal/oauth/oauth.go
Comment (matt-condon) [inline internal/oauth/oauth.go:38]: Q: Can we set "native" by default in [profile.go](https://github.com/mongodb/mongodb-atlas-cli/blob/master/internal/config/profile.go#L76)?

Then conditionally replace it in this file?

File: internal/oauth/oauth.go
Comment (wtrocki) [inline internal/oauth/oauth.go:68]: FYI Config values can be overridden by existing env vars:
https://www.mongodb.com/docs/atlas/cli/stable/atlas-cli-env-variables/

We do not document user agent but it might make sense to use this mechanism?

File: Dockerfile
Comment (wtrocki) [reply Dockerfile:3]: Generic name with no value set. 

```suggestion
ENV MONGODB_ATLAS_USER_AGENT="atlascli-container"
```

Comment (nicohrubec) [reply Dockerfile:3]: I agree that the naming of the var could be improved. I think setting a different value does not really make sense though, because the variable is used as a flag and the value is never used. I set it to an empty string only because env vars in Docker cannot be undefined. (At least I did not find a way)

Comment (nicohrubec) [reply Dockerfile]: I updated the variable name to be more descriptive, lmk what you think

Comment (wtrocki) [reply Dockerfile]: My point is that this PR could be just an `MONGODB_ATLAS_USER_AGENT="atlascli-container"` 
Variable name cannot be changed in this case as it is comming from documented feature: https://www.mongodb.com/docs/atlas/cli/stable/atlas-cli-env-variables/

Comment (wtrocki) [reply Dockerfile]: Feel free to not use that feature (not blocking anything in this PR). 
My only strong opinion is to avoid empty variables. If you need flag like now:

```
MONGODB_ATLAS_RUNNING_FROM_CONTAINER=true
```

One could argue that it introduces some level of proliferation. 
We should likely look into what we want is the effect:

For example disabling atlas cli to run any external binaries 
```
MONGODB_ATLAS_DISABLE_EXTERNAL_COMMANDS=true
MONGODB_ATLAS_USER_AGENT="atlascli-container"
```

This is not only easier to implement and maintain. It does fit well with existing features and config.
Each of those ideally should be connected with the config and documented so users will be aware of the differences when running container etc.



CONTAINER or other values are subject

File: internal/oauth/oauth.go
Comment (nicohrubec) [reply internal/oauth/oauth.go]: Seems reasonable I changed the approach

File: Dockerfile
Comment (nicohrubec) [reply Dockerfile]: MONGODB_ATLAS_USER_AGENT="atlascli-container" doesn't seem to change the UserAgent in my container. So I changed the approach to set default in profile and then replace in case the new env var is set to true as suggested

**Comment by nicohrubec** [comment]
Waiting for feedback from Jakub before merging

**Comment by nicohrubec** [comment]
Approach got approved from Data Analytics

**Review by matt-condon — APPROVED** [review]
LGTM, nice work!

**Pull request merged**
